src/Components/Browser.JS/src/Boot.Server.ts

```diff
@@ -95,7 +95,16 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
     circuits.push(circuit);
   }
 
-  await reconnectTask;
+  if (!await reconnectTask) {
+    // reconnection for the initialConnection will only fail when the application is prerendered but the server no longer has state for the
+    // Circuit. One fairly trivial way to get in to this state is to have the browser reopen a closed tab with the initial render rehydrated
+    // from cache.
+    // In this event, the best thing to try would be to attempt reloading the page.
+    console.assert(!renderingFailed, 'Initial connection should not have encountered a rendering failure.');
+    console.assert(circuits.length > 0, 'Initial render should only fail if there are circuits available.');
```

Comment:
I'm not so sure about using `console.assert` in this kind of way. These messages make sense to *us*, but they are phrased very oddly from the point of view of developers using the framework. I think customers would get on better with messages phrased like:

```
Initial connection failed during rendering.
```

or

```
Initial connection failed, even though no prerendered components were discovered in the page.
```

These still aren't super useful messages (it doesn't provide any indication about what you could do to fix it) but at least it's more direct about saying what the problem is.

Reply:
I also wouldn't use console.xxx directly. We have a logger for these things.